adapters/appnexus/appnexus.go

```diff
@@ -345,6 +354,16 @@ func (a *AppNexusAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters
 			errs = append(errs, err)
 			return nil, errs
 		}
+	} else if isAMP != 0 {
+		// if isAMP is 0, bidder doesn't need it included
+		var err error
+		reqExt.Appnexus = &appnexusReqExtAppnexus{}
+		reqExt.Appnexus.IsAMP = isAMP
+		request.Ext, err = json.Marshal(reqExt)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
```

Comment:
If the entire point of `isAmp` is to serve the purpose of a boolean value, can't we do so? We can declare it in the struct to not be a pointer so its default value is `false` and not `nil`. That is, unless there is an additional purpose `isAMP` serves that makes it absolutely neccessary for us to use an `int`.
```
   94
   95 type appnexusReqExtAppnexus struct {
   96     IncludeBrandCategory    *bool `json:"include_brand_category,omitempty"`
   97     BrandCategoryUniqueness *bool `json:"brand_category_uniqueness,omitempty"`
   98     IsAMP                   bool `json:"brand_category_uniqueness"`  //<-- declare it as bool so it defaults to false and not nil
   99 }
  100
adapters/appnexus/appnexus.go [+][RO]
```

Reply:
This is driven by the requirement from impbus, it is not a decision Joe made.